src/app/search-list/service/search-list.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { BaseService } from '../../core/base';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchListService extends BaseService {
+
+  getMovies(title, searchByKey, searchYear, pageNo?) {
+    const url = searchYear ? `${searchByKey}=${title}*&y=${searchYear}` : `${searchByKey}=${title}*`;
+    return this.get(url)
+      .pipe(map(data => {
+        return data;
+      }));
+  }
+  getNextMovies(title, searchByKey, searchYear, pageNo) {
+    const url = searchYear ?  `${searchByKey}=${title}*&page=${pageNo}&y=${searchYear}` : `${searchByKey}=${title}* `;
```

Comment:
When the year value is not set, the same results are repeated as the user scrolls down.
Did you make a mistake here?